packages/runtime/runtime-definitions/src/id-compressor/idCompressor.ts

```diff
@@ -0,0 +1,140 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+	SessionSpaceCompressedId,
+	OpSpaceCompressedId,
+	SessionId,
+	FinalCompressedId,
+	StableId,
+} from "./identifiers";
+import {
+	IdCreationRange,
+	SerializedIdCompressor,
+	SerializedIdCompressorWithNoSession,
+	SerializedIdCompressorWithOngoingSession,
+} from "./persisted-types";
+
+/**
+ * This is the set of functions on an IdCompressor that a DDS author shouldn't be concerned with.
+ * Operations here are automatically handled at the SharedObject level.
+ */
+export interface IIdCompressorCore {
+	/**
+	 * Finalizes the supplied range of IDs (which may be from either a remote or local session).
+	 * @param range - the range of session-local IDs to finalize.
+	 */
+	finalizeCreationRange(range: IdCreationRange): void;
+
+	/**
+	 * Returns a range of local IDs created by this session in a format for sending to the server for finalizing.
+	 * The range will include all local IDs generated via calls to `generateCompressedId` since the last time this method was called.
+	 * @returns the range of session-local IDs, which may be empty. This range must be sent to the server for ordering before
+	 * it is finalized. Ranges must be sent to the server in the order that they are taken via calls to this method.
+	 */
+	takeNextCreationRange(): IdCreationRange;
+
+	/**
+	 * Returns an iterable of all IDs created by this compressor.
+	 */
+	getAllIdsFromLocalSession(): IterableIterator<SessionSpaceCompressedId>;
+
+	/**
+	 * Returns a persistable form of the current state of this `IdCompressor` which can be rehydrated via `IdCompressor.deserialize()`.
+	 * This includes finalized state as well as un-finalized state and is therefore suitable for use in offline scenarios.
+	 */
+	serialize(
+		withSession: boolean,
+	): SerializedIdCompressorWithOngoingSession | SerializedIdCompressorWithNoSession;
+
+	/**
+	 * Returns a persistable form of the current state of this `IdCompressor` which can be rehydrated via `IdCompressor.deserialize()`.
+	 * This includes finalized state as well as un-finalized state and is therefore suitable for use in offline scenarios.
+	 */
+	serialize(withSession: true): SerializedIdCompressorWithOngoingSession;
+
+	/**
+	 * Returns a persistable form of the current state of this `IdCompressor` which can be rehydrated via `IdCompressor.deserialize()`.
+	 * This only includes finalized state and is therefore suitable for use in summaries.
+	 */
+	serialize(withSession: false): SerializedIdCompressorWithNoSession;
+	serialize(withSession: boolean): SerializedIdCompressor;
+}
+
+/**
+ * This is the set of operations on an IdCompressor that DDS author
```

Comment:
These docs imply the existance of deeper IdCompressor interfaces, which is odd from a consumer perspective. I'd doc it as just the ID compressor: what it does and how you should use it.

Reply:
I've moved the `IdCompressor` docs to this interface and then linked the other docs to the interface